homeassistant/config.py

```diff
@@ -483,6 +514,73 @@ def merge_packages_config(config, packages):
     return config
 
 
+def log_exception(ex, domain, config, hass):
+    """Generate log exception for config validation."""
+    run_callback_threadsafe(
+        hass.loop, async_log_exception, ex, domain, config, hass).result()
+
+
+@callback
+def async_extract_component_config(hass, config, domain):
```

Comment:
I think this should be called `async_check_component_config`. As it returns the whole config, it doesn't extract the component config.